### ISSUE TITLE
What's the final loss at the end of all epoches?

### ISSUE THREAD

**Issue by zidonghua2018** [ml-struct-bio/drgnai#25]
I trained cryodrgn and drgnai using dataset empiar10076, but the loss seemed not to decrease. The losses are 0.655 for cryodrgn and 1.022 for drgnai, but after a few epoches, the loss do not decrease much (0.635 for cryodrgn and 0.99 for drgnai after 3 epoches)

**Comment by zidonghua2018** [comment]
(INFO) (reconstruct.py) (14-Oct-25 19:52:37) # =====> SGD Epoch: 30 finished in 0:04:33.036322; total loss = 0.999678
Is that right?

**Comment by Disyxt** [comment]
how long have you cost on pertaining?

**Comment by michal-g** [comment]
Hi, reposting this answer from a similar recent thread in the issues forum for our sibling package [cryoDRGN](https://github.com/ml-struct-bio/cryodrgn) as it is also relevant here:

> ...changes in loss between training epochs do not necessarily correlate with updates to reconstruction quality nor model convergence due largely to the high-noise regime under which cryo-EM reconstruction models operate. We recommend instead comparing the outputs of the models themselves, as loss can indeed stop decreasing by significant amounts well before the model has stopped making significant improvements to these outputs!
> 
> Have you looked at the output volumes — and if doing heterogeneous reconstruction, the latent spaces — produced by tools such as `analyze` on various epochs from your models? This should give a much better idea of how your models are performing than only looking at the training loss! FYI, generally running cryoDRGN on the full 50S dataset takes at least 5-10 epochs to start converging.
> 
> Let us know if you have any other questions!
>  

 _Originally posted by @michal-g in [#459](https://github.com/ml-struct-bio/cryodrgn/issues/459#issuecomment-3445135039)_

Please note as well that the models implemented by cryoDRGN and cryoDRGN-AI are sufficiently different from one another that comparisons in training loss between the two are unlikely to have a meaningul interpretation.

Let us know if you have any other questions,
Mike
